pay-api/src/pay_api/models/statement.py

```diff
@@ -115,14 +116,49 @@ def payment_methods(self):
 
         return list(payment_methods)
 
+    @hybrid_property
```

Comment:
Have you tested the performance scaling of this property over quite a few statements?

Reply:
Removed this from hybrid property and into the service. The hybrid property was no longer suitable due to the circular dependency issues.

Tested this on a data load with an account over 1000 statements at once, seemed performant